v2/manager.go

```diff
@@ -643,7 +666,17 @@ func (c *Manager) waitForEvents(ec chan<- Event, errCh chan<- error) {
 				}
 				ec <- e
 			} else {
-				errCh <- err
+				// When cgroup is deleted read may return -ENODEV instead of -ENOENT from open
+				_, statErr := os.Lstat(filepath.Join(c.path, "memory.events"))
```

Comment:
Maybe we can use 

If !os.IsNotExist() {
    errCh <- err
}

And one question here: why not just call the close channel in defer?

Reply:
I rewrote it as
```
if  !os.IsNotExist(statErr) {
   errCh <- err
}
```
And added the defer channel close (but its the first line in the function so that it happens with every return.